alibi_detect/utils/pytorch/prediction.py

```diff
@@ -0,0 +1,104 @@
+import numpy as np
+import torch
+import torch.nn as nn
+from typing import Callable, Union
+
+
+def predict_batch(x: Union[np.ndarray, torch.Tensor], model: Union[nn.Module, nn.Sequential],
+                  device: torch.device = None, batch_size: int = int(1e10),
+                  dtype: type = np.float32) -> Union[np.ndarray, torch.Tensor]:
```

Comment:
I don't think `type` is the correct typehint, because `torch.float32` and `tf.float32` are not of type `type`, can't we use `Union[np.float32, torch.float32]` (can make a type alias at the top to make it more succint if more floating point types are to be supported).